plugins/module_utils/foreman_helper.py

```diff
@@ -325,9 +325,10 @@ def run(self, **kwargs):
                 ak_param = {'name': 'kt_activation_keys', 'parameter_type': 'string', 'value': self.foreman_params.pop('activation_keys')}
                 self.foreman_params['parameters'] = parameters + [ak_param]
             elif 'parameters' in self.foreman_params and entity is not None:
-                ak_param = next((param for param in entity.get('parameters') if param['name'] == 'kt_activation_keys'), None)
-                if ak_param:
-                    self.foreman_params['parameters'].append(ak_param)
+                entity_ak_param = next((param for param in entity.get('parameters') if param['name'] == 'kt_activation_keys'), None)
+                user_ak_param = next((param for param in self.foreman_params['parameters'] if param['name'] == 'kt_activation_keys'), None)
```

Comment:
we used "current" and "desired" as prefixes in other places.

Reply:
See, that's what two weeks of PTO do to your brain.